cooperator/models/product.py

```diff
@@ -18,6 +28,7 @@ class ProductTemplate(models.Model):
     by_company = fields.Boolean(string="Can be subscribed by companies?")
     by_individual = fields.Boolean(string="Can be subscribed by individuals?")
     mail_template = fields.Many2one("mail.template", string="Mail template")
+    company_id = fields.Many2one("res.company", default=_default_company_id)
```

Comment:
This will also affect "non share" products, right? Not an issue as far as we can always be able to empty the `company_id` field to make it available to other companies too.

Reply:
as you can see in the `_default_company_id()` method, it depends on the `default_is_share` context value, so it should not affect non-share products, as it returns `False` otherwise.

Reply:
This overwrites the default assignation of company_id for all products, and only assigns company_id by default for share products. This makes that if some other module that assigns company_id by default won't work.

I think this will break products in a multi-company environment. We should see if there is a default for company_id that we don't want to overwrite.

Reply:
Maybe we can assign `company_id` in the on change of the `is_share` boolean? This will make that only in the creation of shares will assign `company_id` and not all cases.

Reply:
From `stock`:

```xml
        <field name="context">{'search_default_group_by_product': 1, 'display_complete': True, 'default_company_id': allowed_company_ids[0]}</field>
```

`allowed_company_ids[0]` appears to be analogous to `self.env.company`. We could give that a spin, and get rid of the default function.

Reply:
@carmenbianca thanks! this is exactly what i was searching for but couldn’t find it. changed to use this and removed the default in the class.